core/js/src/main/scala/com/avsystem/commons/serialization/nativejs/NativeJsonOutput.scala

```diff
@@ -0,0 +1,98 @@
+package com.avsystem.commons
+package serialization.nativejs
+
+import com.avsystem.commons.serialization._
+import com.avsystem.commons.serialization.json.RawJson
+
+import scala.scalajs.js
+import scala.scalajs.js.JSON
+
+final class NativeJsonOutput(
+  valueConsumer: js.Any => Unit,
+  options: NativeFormatOptions,
+) extends OutputAndSimpleOutput {
+
+  override def writeNull(): Unit =
+    valueConsumer(null)
+
+  override def writeString(str: String): Unit =
+    valueConsumer(str)
+
+  override def writeDouble(double: Double): Unit =
+    valueConsumer(double)
+
+  override def writeInt(int: Int): Unit =
+    valueConsumer(int)
+
+  override def writeLong(long: Long): Unit = options.longFormat match {
+    case NativeLongFormat.RawString => writeString(long.toString)
+    case NativeLongFormat.JsNumber => writeDouble(long.toDouble)
+    case NativeLongFormat.JsBigInt => writeRaw(js.BigInt(long.toString))
+  }
+
+  override def writeBigInt(bigInt: BigInt): Unit =
+    writeString(bigInt.toString)
```

Comment:
We customize `Long` format, but not `BigInt` format?

Reply:
Added customization for `BigInt` with two options